gapis/resolve/set.go

```diff
@@ -361,6 +291,71 @@ func change(ctx context.Context, a arena.Arena, p path.Node, val interface{}, r
 	return nil, fmt.Errorf("Unknown path type %T", p)
 }
 
+func changeResources(ctx context.Context, a arena.Arena, after *path.Command, ids []*path.ID, data []*api.ResourceData, r *path.ResolveConfig) (*path.Capture, error) {
+	meta, err := ResourceMeta(ctx, ids, after, r)
+	if err != nil {
+		return nil, err
+	}
+	if len(meta.Resources) != len(ids) {
+		return nil, fmt.Errorf("Expected %d resource(s), got %d", len(ids), len(meta.Resources))
+	}
+
+	cmdIdx := after.Indices[0]
+	// If we change resource data, subcommands do not affect this, so change
+	// the main command.
+
+	oldCmds, err := NCmds(ctx, after.Capture, cmdIdx+1)
+	if err != nil {
+		return nil, err
+	}
+
+	cmds := make([]api.Cmd, len(oldCmds))
+	copy(cmds, oldCmds)
+
+	replaceCommands := func(where uint64, with interface{}) {
+		cmds[where] = with.(api.Cmd)
+	}
+
+	oldCapt, err := capture.ResolveFromPath(ctx, after.Capture)
+	if err != nil {
+		return nil, err
+	}
+	var initialState *capture.InitialState
+	mutateInitialState := func(API api.API) api.State {
+		if initialState == nil {
```

Comment:
Do we think we can split this out into a method. CloneState() I think we will need this more.

Reply:
Done.